src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.cs

```diff
@@ -162,12 +161,46 @@ public Assembly LoadFromPath(string originalAnalyzerPath)
                 assemblyName = null;
             }
 
+            lock (_guard)
+            {
+                _analyzerAssemblyInfoMap[originalAnalyzerPath] = (assemblyName, realPath, SatelliteCultureNames: null);
```

Comment:
Consider the following: 

- Thread1: calls `GetAssemblyInfoForPath` and gets to line 161 and pauses
- Thread2: calls `GetAssemlbyInfoForPath`, completes, then calls `GetSatelliteAssemblyInfoForPath` and populatse culture names
- Thread1: resumes, moves into the `lock` and then blows away the `SatelliteCultureNames` value that Thread2 wrote

Before the write to an entry in the map was one shot so two threads racing didn't make much different (they wrote the same info). Now that the info is written in two stages I think we need to be a bit smarter about it. Maybe simpler to just have a separate table for satellin info.

Reply:
Ha, that's how I first coded it up and I considered it duplicative and changed to this.

I'll go ahead and switch back to multiple dictionaries. 

Alternatively, I think I could also just have a check if originalAnalyzerPath is in the _analyzerAssemblyInfoMap right here and that would also work, but like you said, it's harder to reason about when there are multiple write locations.